example/src/main/java/com/muellerwulff/viewmodelextensions/example/MainModel.kt

```diff
@@ -33,7 +34,7 @@ class MainModel(
         numberOfGenerations++
         if (numberOfGenerations > 5) {
             numberOfGenerations = 0
-            _warning.value = Event("did some generations")
+            _warning.value = "did some generations".asEvent()
```

Comment:
cool idea.

Before i merge this, what do you think of a eventOf() factory function, like mutableLiveDataOf?

Reply:
My goal when creating this was to prevent those nested constructors like f.e. `_result.postValue(Event(Success(mydata)))` and improve readability. I am not sure if this readability could be achieved with a factory method.

Reply:
ok let's compare,

before it was 

`_result.postValue(Event(Success(mydata)))`

and afterwards with the extension it would be:

`_result.postValue(Success(mydata).asEvent())`

and with the factory function it would be:

`_result.postValue(eventOf(Success(mydata)))`

you are probably right, the factory function would do no good here.

mmh I think there would be another way to make this a bit more concise is to add an extension to a `MutableLiveData<Event<T>>`, like: `MutableLiveData<Event<T>>.postValue(value: T)` this call could then call the event foo.

But this would just overcomplicate things. I think we just stay with your idea, which is neat